src/cbmc/cbmc_solvers.cpp

```diff
@@ -237,20 +241,42 @@ std::unique_ptr<cbmc_solverst::solvert> cbmc_solverst::get_smt2(
 
 void cbmc_solverst::no_beautification()
 {
-  if(options.get_bool_option("beautify"))
+  bool beautify = options.get_bool_option("beautify");
+
+  if(beautify)
   {
-    error() << "sorry, this solver does not support beautification" << eom;
-    throw 0;
+    throw invalid_user_input_exceptiont(
+      "Reason: sorry, this solver does not support beautification",
+      "--beautify",
+      "");
   }
 }
 
 void cbmc_solverst::no_incremental_check()
 {
-  if(options.get_bool_option("all-properties") ||
-     options.get_option("cover")!="" ||
-     options.get_option("incremental-check")!="")
+  bool all_properties = options.get_bool_option("all-properties");
+  bool cover = options.get_option("cover") != "";
+  bool incremental_check = options.get_option("incremental-check") != "";
+
+  if(all_properties)
+  {
+    throw invalid_user_input_exceptiont(
+      "Reason: sorry, this solver does not support incremental solving",
+      "--all_properties",
+      "");
+  }
+  else if(cover)
+  {
+    throw invalid_user_input_exceptiont(
+      "Reason: sorry, this solver does not support incremental solving",
+      "--cover",
+      "");
+  }
+  else if(incremental_check)
   {
-    error() << "sorry, this solver does not support incremental solving" << eom;
-    throw 0;
+    throw invalid_user_input_exceptiont(
+      "Reason: sorry, this solver does not support incremental solving",
+      "--incremental-check",
+      "");
```

Comment:
This is quite a big improvement over the previous version that didn't tell you what option it didn't like, :+1: